src/views/Teams/TeamsDetails.js

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react';
+import { fetchTeamsById } from '../../services/fetchteams';
+import { useParams } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import './TeamsDetails.css';
+
+export default function TeamsDetails() {
+  const params = useParams();
+  const [teamdeets, setTeamDeets] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await fetchTeamsById(params.id);
+
+      setTeamDeets(data);
+    };
+    fetchData();
+  }, [params.id]);
+
+  if (!teamdeets) return <div>...Loading</div>;
```

Comment:
Great job preventing the race condition between data mounting and displaying with this loading component!